utility.lua

```diff
@@ -996,6 +996,118 @@ function Auxiliary.FOperationFunRep(f,cc,insf)
 				Duel.SetFusionMaterial(g1)
 			end
 end
+--Fusion monster, name + condition * minc to maxc
+--may not be perfect for gc
+function Auxiliary.AddFusionProcCodeFunRep(c,code,f,minc,maxc,sub,insf)
+	if c.material_count==nil then
+		local code=c:GetOriginalCode()
+		local mt=_G["c" .. code]
+		mt.material_count=1
+		mt.material={code}
+	end
+	local e1=Effect.CreateEffect(c)
+	e1:SetType(EFFECT_TYPE_SINGLE)
+	e1:SetProperty(EFFECT_FLAG_CANNOT_DISABLE+EFFECT_FLAG_UNCOPYABLE)
+	e1:SetCode(EFFECT_FUSION_MATERIAL)
+	e1:SetCondition(Auxiliary.FConditionCodeFunRep(code,f,minc,maxc,sub,insf))
+	e1:SetOperation(Auxiliary.FOperationCodeFunRep(code,f,minc,maxc,sub,insf))
+	c:RegisterEffect(e1)
+end
+function Auxiliary.FConditionFilterCFR(c,code,sub,f,mg,minc)
+	return (c:IsCode(code) or (sub and c:IsHasEffect(EFFECT_FUSION_SUBSTITUTE))) and mg:IsExists(f,minc,c)
+end
+function Auxiliary.FConditionCodeFunRep(code,f,minc,maxc,sub,insf)
+	return	function(e,g,gc,chkf)
+			if g==nil then return insf end
+			local mg=g:Filter(Card.IsCanBeFusionMaterial,nil,e:GetHandler())
+			if gc then return (gc:IsCode(code) or sub and gc:IsHasEffect(EFFECT_FUSION_SUBSTITUTE))
+				and mg:IsExists(f,minc,gc)
+			end
+			local mg1=mg:Filter(aux.FConditionFilterCFR,nil,code,sub,f,mg,minc)
+			local mg2=mg:Filter(f,nil,minc)
+			if chkf~=PLAYER_NONE then return mg1:GetCount()>=1
+				and mg1:FilterCount(Card.IsOnField,nil)+mg2:FilterCount(Card.IsOnField,nil)~=0
+			else return mg1:GetCount()>=1 end
+		end
+end
+function Auxiliary.FOperationCodeFunRep(code,f,minc,maxc,sub,insf)
+	return 	function(e,tp,eg,ep,ev,re,r,rp,gc,chkf)
```

Comment:
is there any problems in https://github.com/Fluorohydride/ygopro-core/pull/11 to filter IsCanBeFusionMaterial in card::fusion_select?